inference-engine/tests/functional/inference_engine/CMakeLists.txt

```diff
@@ -5,25 +5,40 @@
 
 set(TARGET_NAME ieFuncTests)
 
+set(INCLUDES ${IE_MAIN_SOURCE_DIR}/src/inference_engine)
+set(LINK_LIBRARIES
+    gmock
+    funcTestUtils
+    ngraphFunctions
+    inference_engine_transformations
+)
+set(DEPENDENCIES
+    mock_engine
+    inference_engine_ir_reader
+    inference_engine_ir_v7_reader
+    template_extension
+)
+
+if (NGRAPH_ONNX_IMPORT_ENABLE)
+    set(INCLUDES ${INCLUDES} ${OpenVINO_MAIN_SOURCE_DIR}/docs/onnx_custom_op)
+    set(LINK_LIBRARIES ${LINK_LIBRARIES} onnx_importer onnx_custom_op)
+    set(DEPENDENCIES ${DEPENDENCIES} onnx_importer onnx_custom_op)
+else()
+    set(EXCLUDED_SOURCE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/onnx_reader)
+endif()
+
+if (NOT NGRAPH_ONNX_IMPORT_ENABLE OR NOT ENABLE_MKL_DNN)
+    set(EXCLUDED_SOURCE_DIRS ${EXCLUDED_SOURCE_DIRS} ${CMAKE_CURRENT_SOURCE_DIR}/extension)
```

Comment:
can we put extensions.cpp to inference_engine folder directly? I think we don't have to introduce a folder extension with a single file.

Reply:
addIeTargetTest allows us to exclude directories, but not single files.

Reply:
let's fix it. `addIeTargetTest` uses regex which can be simply extended for files as well

Reply:
BTW, `ENABLE_MKL_DNN` can be replaced by runtime check via `Core::GetAvailableDevices()` or `Core::GetMetric("CPU", METRIC_KEY(AVAILABLE_DEVICES))`